api/internal/git/repospec.go

```diff
@@ -118,6 +119,20 @@ func parseGitURL(n string) (
 		path, gitRef, gitTimeout, gitSubmodules = peelQuery(n[index+len(gitDelimiter)+len(orgRepo):])
 		return
 	}
+	if strings.HasPrefix(n, fileScheme) {
+		idx := strings.Index(n, gitSuffix)
+		if idx == -1 {
+			// invalid, must have .git somewhere
```

Comment:
I was not aware of the `//` syntax. I didn't know that there were docs for remoteBuild in the examples directory. Most of my changes here were based on the assumption we didn't have a way to indicate where the repo starts in the URL. I will change to use the `//` syntax.

I do feel that `//` is unintuitive and strange. It's also not documented on the kustomize site as far as I can tell. That's where I go look for docs. In past experience, `//` in a URL is usually a mistake when a developer forgets to use something like `url.Join()` to join URL segments.

Reply:
Btw, all this WEIRDNESS is why I have the following comment in the PR description

> I made a few small changes to repospec to support it. It's a bit hacked but to do it better would require a lot larger change that I do not want to do in this PR.

I think to make this more clean in a way you and I would both like would actually require rewriting this file that I do not wish to do right now. There are a bunch of rarely used, documented features like `_git`, `//`, `git::` that were added for very specific cases but implemented a bit more generically (or not) so that it makes Kustomize be a little more lenient or stricter than it should in certain cases not exercised in tests.

Reply:
>  I didn't know that there were docs for remoteBuild in the examples directory.

Yeah, unfortunately our documentation is quite fragmented: #4338 . I'm eager to support contributors to work on that project, which has quite a low barrier to entry (with limited time, Natasha and I focus on high context work newer folks cannot do).

> Kustomize does not actually have code that supports the // syntax

Implementation aside, support for it does seem deliberate. The test in RepoSpec was specifically added as regression coverage: https://github.com/kubernetes-sigs/kustomize/pull/780. 

> So, I thought at the time you were just using it for documentation in tests. Could you please how you'd like to proceed?

We need to keep supporting `//` both because it is documented and just because it currently works. Whatever has worked has been working for a long time, and we don't want to cause end user pain without a very good reason. FWIW the preamble to one of the RepoSpec tests alludes to this:
```go
func TestNewRepoSpecFromUrl_Permute(t *testing.T) {
	// Generate all many permutations of host, orgRepo, pathName, and ref.
	// Not all combinations make sense, but the parsing is very permissive and
	// we probably stil don't want to break backwards compatibility for things
	// that are unintentionally supported.
```

As you've discovered, the `//` convention actually comes in handy in situations where the `.git` suffix is not used. I tried to find the source of it, and my educated guess is that we inherited it from go-getter: https://github.com/hashicorp/go-getter#subdirectories. Before my time, Kustomize used to use go-getter directly, but it was later replaced with bespoke code because `kubernetes/kubernetes` (for `kubectl kustomize`) could not accept that dependency.

> Btw, all this WEIRDNESS!! is why I have the following comment in the PR description

Thank you for sticking with us through this. We don't have all the historical context on this either, so perhaps that helps with understanding why we felt we needed the code freeze until our coverage of this area is reliable. 😄 

> I think to make this more clean in a way you and I would both like would actually require rewriting this file that I do not wish to do right now.

I totally agree with you that we should save refactoring for after we're finished with the test coverage improvements, and do it separately. If you have ideas for how that should look, we're definitely interested in the follow up PR!

Reply:
> We need to keep supporting // both because it is documented and just because it currently works.

Regarding that, I was saying that it doesn't currently work. It _only_ works in cases where a single `/` would also work just fine.